app/services/battle_service.rb

```diff
@@ -162,6 +162,41 @@ def unready_player(battle_id)
     end
   end
 
+  def kick_player(battle_id, kick_player_id)
+    # Validate caller is admin
+    raise BattleValidationError.new "Player #{@player} is not an admin" unless @player.admin?
+
+    # Validate battle exists
+    battle = validate_battle(battle_id)
+
+    battle.with_lock do
+
+      # Validate the player is in the battle
+      validate_other_player_in_battle(battle, kick_player_id)
```

Comment:
Have you tested this? This looks to be passing an id to the method, but the method seems to expect a player model object. 

Instead of this, let's just use `validate_player_in_battle(battle)` but modify it to accept any player id.

Reply:
it did work though, but better to just have one method